src/js/components/settings/billing.js

```diff
@@ -0,0 +1,228 @@
+import React from 'react';
+import moment from 'moment';
+
+// material ui
+import { Button, Divider, LinearProgress, Tooltip } from '@material-ui/core';
+import WarningIcon from '@material-ui/icons/Warning';
+
+import AppStore from '../../stores/app-store';
+import AppActions from '../../actions/app-actions';
+import MonthlyBillingInformation from './monthlybillinginformation';
+import PlanNotification from './plannotification';
+
+const accuracy = 4;
+const totalFreeCredit = 120;
+
+const interactionList = [
+  {
+    title: 'Devices',
+    billingInformation: [
+      {
+        title: 'Device activations:',
+        key: 'activation',
+        explanation: '# new devices connected to the server'
+      },
+      {
+        title: 'Total active devices:',
+        key: 'monthly_device',
+        explanation: '# devices that were active during this period'
+      }
+    ]
+  },
+  {
+    title: 'Deployments',
+    billingInformation: [
+      {
+        title: 'Device deployments:',
+        key: 'device_deployment',
+        explanation: '# devices targeted by a deployment'
+      }
+    ]
+  },
+  {
+    title: 'Data storage',
+    billingInformation: [
+      {
+        title: 'GB of data stored:',
+        key: 'data_storage',
+        explanation: 'The software you have uploaded to the hosted Mender service'
+      }
+    ]
+  },
+  {
+    title: 'Data transfer',
+    billingInformation: [
+      {
+        title: 'GB of data transferred:',
+        key: 'data_transfer',
+        explanation: 'Data traffic that was sent between a device the hosted Mender service'
+      }
+    ]
+  }
+];
+
+class Billing extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      showUsage: false,
+      billingInformation: {
+        interactions: [],
+        timeframe: { month: 0, year: 0 },
+        timestamp: new Date(),
+        total: 0
+      }
+    };
+  }
+
+  componentDidMount() {
+    this.updateBillingStatement();
+  }
+
+  updateBillingStatement(date = new Date()) {
+    const self = this;
+    const timeframe = { month: date.getMonth(), year: date.getFullYear() };
+    AppActions.getBillingStatement(timeframe).then(information => {
+      const interactions = interactionList.map(item => {
+        item.billingInformation = item.billingInformation.map(infoItem => ({
+          ...infoItem,
+          unitFee: parseFloat(information[`${infoItem.key}_unit_fee`].toFixed(accuracy)),
+          total: parseFloat(information[`${infoItem.key}_total_cost`].toFixed(accuracy)),
+          quantity: parseFloat(information[`${infoItem.key}_usage`].toFixed(accuracy))
+        }));
+        return item;
+      });
+
+      self.setState({
+        billingInformation: {
+          interactions,
+          timeframe,
+          timestamp: new Date(information.timestamp),
+          total: information.summary_total_cost
```

Comment:
This should be fixed to 2 decimal points

Reply:
argh 🙈 - should be fixed now